src/main/java/ooss/Teacher.java

```diff
@@ -15,11 +16,23 @@ public Teacher (int id, String name, int age){
 
     @Override
     public String introduce() {
-        return "My name is " + name + ". I am " + age + " years old. I am a teacher.";
+        if (klass != null) {
+            return "My name is " + name + ". I am " + age + " years old. I am a teacher. I am in class " + klass.getClassNumber() + ".";
+        } else {
+            return "My name is " + name + ". I am " + age + " years old. I am a teacher.";
+        }
     }
 
+    public boolean belongsTo(Klass klass) {
```

Comment:
klass should be arraylist, use stream api to match the input klass and exist klass .